### DISCUSSION TITLE
Help Drop event action not fire

### DISCUSSION THREAD

**Discussion by ghost** [canton7/Stylet#332] [Q&A] [answered]
I need the drop event on a window , so according to : https://github.com/canton7/Stylet/wiki/Actions I put this in XAML:

`AllowDrop="True" Drop="{s:Action DroppedSomething}"`
and  in the viewmodel:
 ```
Private Sub DroppedSomething(sender As Object, e As DragEventArgs)
       Debug.WriteLine("DroppedSomething")
    End Sub
```
but it never fire.
If I change it to:
```
 Private Sub DroppedSomething()
      Debug.WriteLine("DroppedSomething")
        End If
    End Sub
```

it fire but I can't get the objects.
here the code in c#
```
private void DroppedSomething(object sender, DragEventArgs e)
{
    Debug.WriteLine("DroppedSomething");
}
```

and 
```
public void DroppedSomething()
{
    Debug.WriteLine("DroppedSomething");
}
```

what am I missing?

**Comment by canton7** [comment, marked as answer]
I can't reproduce this in a simple test: both `public void DoSomething()` and `public void DroppedSomething(object sender, DragEventArgs e)` work as expected.

Note that your method has to be public (and yours are private) -- you'll get an exception if this isn't the case. But that should be that case for both signatures.
